File: DONGUK/프로그래머스_WEEK_10/n진수 게임.py
def convert(num, base):
    tmp = '0123456789ABCDEF'
    q = num // base
    r = num % base

    ret_str = tmp[r]

    if q == 0:
        return tmp[r]

    while q > 0:
        r = q % base
        q = q // base
        ret_str += tmp[r]

    ret_str = ret_str[::-1]
    return ret_str


def solution(n, t, m, p):
    answer = ''
    allCase = ''

    myTurn = []
    for i in range(m*t): # 숫자는 0부터 시작
        allCase += convert(i,n)

        if i == p-1: # 순서는 1부터 시작
            myTurn.append(i)
            p += m
    
    for turn in myTurn:
        answer += allCase[turn]

    return answer

File: KIJUNG/PRO_WEEK_06/n진수게임.cpp
#include <string>
#include <math.h>

using namespace std;

string e = "0123456789ABCDEF";

string trans(int notation,int num){
    string tmp = "";
    int p = 0;
    
    while(num>=(int)pow(notation,p)) p++;
    if(p>0) p--;

    while(p>=0){
        int q = num/(int)pow(notation, p);
        tmp+=e[q];
        num = num%(int)pow(notation, p);
        p--;
    }

    return tmp;
}

string solution(int n, int t, int m, int p) {
    string answer = "";
    string seq = "0";

    //진수 변환
    for(int i=1;seq.size()<m*t;i++) seq += trans(n, i);
    for(int i=0;i<t;i++) answer += seq[i*m+p-1];
    
    return answer;
}